mediapipe/tasks/python/vision/gesture_recognizer.py

```diff
@@ -0,0 +1,393 @@
+       for hand_world_landmark in hand_world_landmarks.landmark]
+      for hand_world_landmarks in hand_world_landmarks_proto_list
+    ]
+  )
+
+
+@dataclasses.dataclass
+class GestureRecognizerOptions:
+  """Options for the gesture recognizer task.
+
+  Attributes:
+    base_options: Base options for the hand gesture recognizer task.
+    running_mode: The running mode of the task. Default to the image mode.
+      Gesture recognizer task has three running modes:
+      1) The image mode for recognizing hand gestures on single image inputs.
+      2) The video mode for recognizing hand gestures on the decoded frames of a
+         video.
+      3) The live stream mode for recognizing hand gestures on a live stream of
+         input data, such as from camera.
+    num_hands: The maximum number of hands can be detected by the recognizer.
+    min_hand_detection_confidence: The minimum confidence score for the hand
+      detection to be considered successful.
+    min_hand_presence_confidence: The minimum confidence score of hand presence
+      score in the hand landmark detection.
+    min_tracking_confidence: The minimum confidence score for the hand tracking
+      to be considered successful.
+    canned_gesture_classifier_options: Options for configuring the canned
+      gestures classifier, such as score threshold, allow list and deny list of
+      gestures. The categories for canned gesture classifiers are:
+      ["None", "Closed_Fist", "Open_Palm", "Pointing_Up", "Thumb_Down",
+      "Thumb_Up", "Victory", "ILoveYou"]
+      TODO :Note this option is subject to change.
+    custom_gesture_classifier_options: Options for configuring the custom
+      gestures classifier, such as score threshold, allow list and deny list of
+       gestures.
+      TODO :Note this option is subject to change.
+    result_callback: The user-defined result callback for processing live stream
+      data. The result callback should only be specified when the running mode
+      is set to the live stream mode.
+  """
+  base_options: _BaseOptions
+  running_mode: _RunningMode = _RunningMode.IMAGE
+  num_hands: Optional[int] = 1
+  min_hand_detection_confidence: Optional[float] = 0.5
+  min_hand_presence_confidence: Optional[float] = 0.5
+  min_tracking_confidence: Optional[float] = 0.5
+  canned_gesture_classifier_options: Optional[_ClassifierOptions] = _ClassifierOptions()
+  custom_gesture_classifier_options: Optional[_ClassifierOptions] = _ClassifierOptions()
+  result_callback: Optional[
+      Callable[[GestureRecognitionResult, image_module.Image,
+                int], None]] = None
+
+  @doc_controls.do_not_generate_docs
+  def to_pb2(self) -> _GestureRecognizerGraphOptionsProto:
+    """Generates an GestureRecognizerOptions protobuf object."""
+    base_options_proto = self.base_options.to_pb2()
+    base_options_proto.use_stream_mode = False if self.running_mode == _RunningMode.IMAGE else True
```

Comment:
nit: could be simplified to 
`base_options_proto.use_stream_mode = self.running_mode == _RunningMode.IMAGE`